src/storage/v2/edge_accessor.cpp

```diff
@@ -123,6 +123,24 @@ Result<storage::PropertyValue> EdgeAccessor::SetProperty(PropertyId property, co
   return std::move(current_value);
 }
 
+Result<bool> EdgeAccessor::SetProperties(std::map<storage::PropertyId, storage::PropertyValue> &properties) {
+  utils::MemoryTracker::OutOfMemoryExceptionEnabler oom_exception;
+  if (!config_.properties_on_edges) return Error::PROPERTIES_DISABLED;
+
+  std::lock_guard<utils::SpinLock> guard(edge_.ptr->lock);
```

Comment:
sorry for being ignorant, so where is this guard used after initialization, I'm not sure how does this class work

Reply:
I must say I am not sure 100%, but I think with this class you guarantee that you don't work on such edge and that is it.

Reply:
After checking the code, it is exactly what happens, and on the destruction of the object, the spin lock is released.